sandbox-tools/merge-repos/src/syncReposToStaging.ts

```diff
@@ -507,100 +509,222 @@ async function resolveConflictsToTheirs(git: SimpleGit, commitDetails: ICommitDe
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "E", path: destFolder + "/" + destFile } , "Removing extra file");
+                await git.rm(destFolder + "/" + destFile);
             }
-        };
+        }
+    }
+}
+
+async function getAndSyncSrcRepo(git: SimpleGit, repoName: string, details: IRepoDetails): Promise<ISourceRepoDetails> {
+
+    // Get a clone of the source repo and reset to the starting point
+    let srcOriginRepoUrl = details.url;
+    let srcOriginBranch = details.branch;
+
+    let forkDestOrg =  _mergeGitRoot + "-" + repoName;
+
+    // Now lets go and create a local clone
+    log(`Cloning the source repo ${srcOriginRepoUrl} branch ${srcOriginBranch} to ${forkDestOrg}`);
+    await git.clone(srcOriginRepoUrl, forkDestOrg, [ "-b", srcOriginBranch]);
+
+    let orgGit = createGit(forkDestOrg, "merge.org." + repoName + ".git");
+    let checkoutArgs = [
+        "--progress",
+        "-B", srcOriginBranch
+    ];
 
-        commitDetails.message += commitMessage;
+    if (details.branchStartPoint) {
+        // Used for testing the the consistent "merging" over time based on using the configured
+        // tags (startPoints) from the original master repo.
+        checkoutArgs.push(details.branchStartPoint);
     } else {
-        log(` - No files found in ${baseFolder + "/" + srcFolder}`);
+        checkoutArgs.push("HEAD")
     }
+
+    await orgGit.checkout(checkoutArgs);
+
+    // Reset the local branch to the requested HEAD (or mergeStartPoint -- used for testing)
+    log("Resetting...");
+    await orgGit.reset(["--hard"]).catch(abort(git, "Failed to hard reset"));
+
+    // Remove any untracked files in this local branch
+    log("Cleaning...");
+    // The excludes where for local development / branch purposes to ensure local changes where not lost
+    await orgGit.clean([CleanOptions.RECURSIVE, CleanOptions.FORCE], ["-e", "/.vs"]).catch(abort(git, "Failed during clean"));
+
+    let hashDetails = await orgGit.show(["-s", details.branchStartPoint ? details.branchStartPoint : "HEAD"]).catch(abort(git, `Failed getting hash details ${details.branchStartPoint ? details.branchStartPoint : "HEAD"}`));
+
+    let commitHash: string = "";
+    let commitDetails = /^commit\s+(\w+)/g.exec(hashDetails || "");
+    if (commitDetails && commitDetails[1]) {
+        commitHash = commitDetails[1];
+    }
+
+    let ignoreTags: string[] = [];
+    Object.keys(reposToSyncAndMerge).forEach((value) => {
+        ignoreTags.push(reposToSyncAndMerge[value].tagPrefix + "/");
+    });
+
+    // rename the tags from the original repos so they have a prefix and remove the original
+    await renameTags(orgGit, reposToSyncAndMerge, details.tagPrefix + "/", ignoreTags);
+
+    let moveCommitMessage: ICommitDetails = {
+        committed: false,
+        message: `${repoName} @ [${commitHash.substring(0, 7)}...](${details.url}/commit/${commitHash})\n${hashDetails}`
+    }
+
```

Comment:
This now moves all of the files / folders of the local `source` clone what will become the new destination in the `staging` branch (ie. `auto-merge/xxxx`)